src/components/dialogs/AddStaffNotationDialog.vue

```diff
@@ -0,0 +1,236 @@
+  @Prop() readonly dialog: boolean
+
+  /** Prop to provide attachment selector. */
+  @Prop() readonly attach: string
+
+  /** Prop to require court order number regardless the plan of arrangement. */
+  @Prop() readonly courtOrderNumberRequired: boolean
+
+  /** Entity number */
+  readonly entityIncNo!: string
+
+  /** The notation text. */
+  private notation: string = ''
+
+  /** Court Order Number */
+  private courtOrderNumber:string = null
+
+  /** Whether has plan of arrangement */
+  private planOfArrangement = false
+
+  /** Whether this component is currently saving. */
+  private saving = false
+
+  /** Court Order component key */
+  private courtOrderKey = 0
+
+  /** Notation form validity */
+  private notationFormValid = false
+
+  /** Notation max length */
+  private notationMaxLength = 2000
+
+  /** Flag to enable validation in this component */
+  private enableValidation = false
+
+  private get notationRules (): Array<Function> {
+    if (this.enableValidation) {
+      return [
+        (v: string) => !!v || `Enter a ${this.itemName}`,
+        (v: string) => v.length <= this.notationMaxLength || 'Maximum characters exceeded.'
+      ]
+    } else return []
+  }
+
+  private setFileNumber (courtOrderNumber: string): void {
+    this.courtOrderNumber = courtOrderNumber
+  }
+
+  private setHasPlanOfArrangement (planOfArrangement: boolean):void {
+    this.planOfArrangement = planOfArrangement
+  }
+  /** Called when prop changes (ie, dialog is shown/hidden). */
+  @Watch('dialog')
+  private async onDialogChanged (val: boolean): Promise<void> {
+    // when dialog is shown, reset notation and validation
+    if (val) {
+      this.notation = ''
+      await Vue.nextTick()
+      this.$refs.notationFormRef.resetValidation()
+
+      // This will make CourtOrderPoa to re-render as it will have a different key every time it opens the dialog.
+      // Source: https://michaelnthiessen.com/force-re-render/
+      this.courtOrderKey++
+    }
+  }
+
+  /**
+   * Emits event to close this dialog.
+   * @param needReload Whether the dashboard needs to be reloaded.
+   */
+  @Emit('close')
+  private emitClose (needReload: boolean): void {
+    this.enableValidation = false
+  }
+
+  /** WIP */
+  /** Saves the current notation. */
+  private async save (): Promise<void> {
+    // prevent double saving
+    if (this.saving) return
+
+    this.enableValidation = true
+    await this.$nextTick()
+    const isNotationFormRefValid = this.$refs.notationFormRef.validate()
+    const isCourtOrderPoaFormRefValid = this.$refs.courtOrderPoaRef.validate()
+    if (!isNotationFormRefValid || !isCourtOrderPoaFormRefValid) {
+      return
+    }
+
+    this.saving = true
+
+    const data = {
+      'filing': {
+        'header': {
+          'name': this.filingType,
+          'date': this.currentDate
+          // 'certifiedBy': 'full name',
+          // 'email': 'no_one@never.get',
+          // 'filingId': 1
+        },
```

Comment:
This looks incomplete.

Also, if my PR is merged before yours, please add `isStaff: this.isStaffRole` to the header.

Or if your PR is merged before mine, please remind me to add that ;)

Reply:
Yes, it is incomplete. I put a comment on top of the method. I'm still depending on #4390

I'm gonna add isStaff to the header. Thanks.

Reply:
I see we don't need the isStaff as we should trust the JWT token, right? Not adding then.

Reply:
Yes, sorry about that. Finally the design has being discussed. Do not add `isStaff`.